janusgraph-core/src/main/java/org/janusgraph/graphdb/transaction/StandardJanusGraphTx.java

```diff
@@ -697,6 +699,65 @@ private void checkConnectionConstraintOrCreateConnectionConstraint(JanusGraphVer
         config.getAutoSchemaMaker().makeConnectionConstraint(edgeLabel, outVertexLabel, inVertexLabel, this);
     }
 
+    public JanusGraphEdge addEdge(String relationId, JanusGraphVertex outVertex, JanusGraphVertex inVertex, EdgeLabel label){
+        verifyWriteAccess(outVertex, inVertex);
+        outVertex = ((InternalVertex) outVertex).it();
+        inVertex = ((InternalVertex) inVertex).it();
+        Preconditions.checkNotNull(label);
+        checkConnectionConstraintOrCreateConnectionConstraint(outVertex, inVertex, label);
+        Multiplicity multiplicity = label.multiplicity();
+        TransactionLock uniqueLock = getUniquenessLock(outVertex, (InternalRelationType) label,inVertex);
+        if (!graph.getConfiguration().allowEdgeIdSetting()) {
+            throw new UnsupportedOperationException(String.format("Provided relation id [%s] is ignored because edge id setting is not enabled", relationId));
+            // relationId = null;
+        }
+        uniqueLock.lock(LOCK_TIMEOUT);
+        try {
+            //Check uniqueness
```

Comment:
Shouldn't this moved into a different method to reduce duplication.

Reply:
@farodin91, I wrapped the check part in one func to reduce the duplication.